File: dart/resistor-color/lib/resistor_color.dart
class ResistorColor {
  Map<String, int> colorCodes = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
  };

  int colorCode(String color) {
    return colorCodes[color];
  }

  List<String> get colors {
    return colorCodes.keys.toList();
  }
}

File: cpp/resistor-color/resistor_color.cpp
#include "resistor_color.h"

namespace resistor_color
{
    std::vector<std::string> colors()
    {
        return {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    }

    int color_code(const std::string &color)
    {
        std::vector<std::string> all = colors();
        for (size_t i = 0; i < all.size(); i++)
        {
            if (all[i] == color)
                return i;
        }

        return -1;
    }

} // namespace resistor_color
